src/controller.js

```diff
@@ -27,38 +35,148 @@ class ChatbotCtrl {
+          { value: 0, text: 'livechat_survey_answer_unsatisfied', img: 'sad' },
+        ],
+      },
+      {
+        id: 'q3',
+        text: 'livechat_survey_question3',
+        answers: [
+          { value: 4, text: 'chatbot_answer_yes', img: 'happy' },
+          { value: 0, text: 'chatbot_answer_no', img: 'sad' },
+        ],
+      },
+    ];
+
     this.$scope.CHATBOT_MESSAGE_QUALITY = CHATBOT_MESSAGE_QUALITY;
     this.$scope.CHATBOT_MESSAGE_TYPES = CHATBOT_MESSAGE_TYPES;
     this.$scope.CHATBOT_SURVEY_STEPS = CHATBOT_SURVEY_STEPS;
 
     this.$rootScope.$on('ovh-chatbot:open', () => this.open());
     this.$rootScope.$on('ovh-chatbot:opened', () => this.focusInput());
+  }
+
+  init() {
+    this.initialized = true;
 
     this.config = this.config || {};
     if (isString(this.url)) {
       this.config.url = this.url;
     }
     this.ChatbotService.setConfig(this.config);
+
+    if (this.canSwitchToLivechat) {
+      this.initializeLivechat();
+    }
+  }
+
+  initializeLivechat() {
+    this.$translate('livechat_agent_default_name').then((agentName) => {
+      this.agentName = agentName;
+    });
+
+    return this.LivechatService.getConfiguration().then((config) => {
+      this.livechatFactory = new this.LivechatFactory(
+        config,
+        this.config.language,
+        this.config.country,
+        this.bindLivechatCallbacks({
+          onConnectSuccess: this.onLivechatConnectionSuccess,
+          onWelcomeMessage: this.onLivechatWelcome,
+          onAgentMessage: this.onLivechatAgentMessage,
+          onSystemMessage: this.onLivechatSystemMessage,
+          onHistory: this.onLivechatHistory,
+          onSurvey: this.onLivechatSurvey,
+          onConnectionFailure: this.onLivechatConnectionFailure,
+          onError: this.onLivechatError,
+          onNoAgentsAvailable: this.onLivechatNoAgentsAvailable,
+          onAgentStartTyping: this.onLivechatAgentStartTyping,
+          onAgentStopTyping: this.onLivechatAgentStopTyping,
+        }),
+      );
+
+      return this.livechatFactory.restore().then((chatRestored) => {
+        if (chatRestored) {
+          this.started = true;
+          this.livechat = true;
+          this.open();
+        }
+      }).catch(() => {
+        this.onLivechatConnectionFailure();
+        this.started = true;
+        this.open();
+      });
+    }).catch(() => {
+      this.livechatFactory = null;
+    }).finally(() => {
+      this.loaders.isStartingLivechat = false;
+    });
+  }
+
+  bindLivechatCallbacks(callbacks) {
+    const boundCallbacks = {};
+
+    Object.entries(callbacks).forEach(([event, callback]) => {
+      const boundCallback = callback.bind(this);
+
+      // Bind callbacks to instance and scope
+      // with original arguments
+      boundCallbacks[event] = ((...args) => {
+        this.$scope.$apply(() => {
+          boundCallback(...args);
+        });
+      });
+    });
+
+    return boundCallbacks;
```

Comment:
What is the purpose here ? Why do you need to do that ?

Reply:
That's a requirement of the livechat vendor lib (see usage).